samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -233,32 +214,41 @@ public void waitForFinish() {
    * @throws TimeoutException exception for latch timeout
    */
   /* package private */ void createStreams(String planId, List<StreamSpec> intStreams) throws TimeoutException {
-    if (!intStreams.isEmpty()) {
-      // Move the scope of coordination utils within stream creation to address long idle connection problem.
-      // Refer SAMZA-1385 for more details
-      CoordinationUtils coordinationUtils = createCoordinationUtils();
-      if (coordinationUtils != null) {
-        Latch initLatch = coordinationUtils.getLatch(1, planId);
-
-        try {
-          // check if the processor needs to go through leader election and stream creation
-          if (shouldContestInElectionForStreamCreation(initLatch)) {
-            LeaderElector leaderElector = coordinationUtils.getLeaderElector();
-            leaderElector.setLeaderElectorListener(() -> {
-                getStreamManager().createStreams(intStreams);
-                initLatch.countDown();
-              });
-            leaderElector.tryBecomeLeader();
-            initLatch.await(LATCH_TIMEOUT_MINUTES, TimeUnit.MINUTES);
-          }
-        } finally {
-          coordinationUtils.reset();
-        }
-      } else {
-        // each application process will try creating the streams, which
-        // requires stream creation to be idempotent
+    if (intStreams.isEmpty()) {
+      LOG.info("Set of intermediate streams is empty. Nothing to create.");
+      System.out.println("1");
+      return;
+    }
+
+    // Move the scope of coordination utils within stream creation to address long idle connection problem.
+    // Refer SAMZA-1385 for more details
+    String coordinationId = new ApplicationConfig(config).getGlobalAppId() + APPLICATION_RUNNER_ZK_PATH_SUFFIX;
+    CoordinationUtils coordinationUtils =
+        CoordinationUtilsFactory.getCoordinationUtilsFactory(config).getCoordinationUtils(coordinationId, uid, config);
+    System.out.println(" create streams." + uid);
+    if (coordinationUtils == null) {
+      // each application process will try creating the streams, which
+      // requires stream creation to be idempotent
+      getStreamManager().createStreams(intStreams);
+      System.out.println("failed to create utils" + uid);
+      return;
+    }
+
+    DistributedLockWithState lockWithState = coordinationUtils.getLockWithState(planId);
+    System.out.println("Init NEEDED for streams" + uid);
+    try {
+      // check if the processor needs to go through leader election and stream creation
+      if (lockWithState.lockIfNotSet(1000, TimeUnit.MILLISECONDS)) {
+        System.out.println("CREATING STREAMS" + uid);
```

Comment:
probably need some meaningful logging here, ilke log.info("lock {} is acquired", lockId)

Reply:
Done.